internal/api/logs/v1/http.go

```diff
@@ -0,0 +1,149 @@
+package http
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/go-kit/kit/log"
+	"github.com/observatorium/observatorium/internal/proxy"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	ReadTimeout        = 15 * time.Minute
+	WriteTimeout       = time.Minute
+	lokiUpstreamPrefix = "/loki"
+)
+
+type handlerConfiguration struct {
+	logger     log.Logger
+	registry   *prometheus.Registry
+	instrument handlerInstrumenter
+}
+
+// HandlerOption modifies the handler's configuration
+type HandlerOption func(h *handlerConfiguration)
+
+// Logger add a custom logger for the handler to use.
+func Logger(logger log.Logger) HandlerOption {
+	return func(h *handlerConfiguration) {
+		h.logger = logger
+	}
+}
+
+// Registry adds a custom Prometheus registry for the handler to use.
+func Registry(r *prometheus.Registry) HandlerOption {
+	return func(h *handlerConfiguration) {
+		h.registry = r
+	}
+}
+
+// HandlerInstrumenter adds a custom HTTP handler instrument middleware for the handler to use.
+func HandlerInstrumenter(instrumenter handlerInstrumenter) HandlerOption {
+	return func(h *handlerConfiguration) {
+		h.instrument = instrumenter
+	}
+}
+
+type handlerInstrumenter interface {
+	NewHandler(labels prometheus.Labels, handler http.Handler) http.HandlerFunc
+}
+
+type nopInstrumentHandler struct{}
+
+func (n nopInstrumentHandler) NewHandler(labels prometheus.Labels, handler http.Handler) http.HandlerFunc {
+	return handler.ServeHTTP
+}
+
+func NewHandler(read, write *url.URL, opts ...HandlerOption) http.Handler {
+	c := &handlerConfiguration{
+		logger:     log.NewNopLogger(),
+		registry:   prometheus.NewRegistry(),
+		instrument: nopInstrumentHandler{},
+	}
+	for _, o := range opts {
+		o(c)
+	}
+
+	r := chi.NewRouter()
+
+	if read != nil {
+		var proxyRead http.Handler
+		{
+			read.Path = fmt.Sprintf("%s%s", read.Path, lokiUpstreamPrefix)
```

Comment:
let's use `path.Join` here rather than `fmt.Sprintf` just for safety in case anything unexpected is passed